devices/HumanDataCollector/HumanDataCollector.cpp

```diff
@@ -0,0 +1,663 @@
+        pImpl->baseVelocityDataPort.write(true);
+        pImpl->stateJointNamesDataPort.write(true);
+        pImpl->jointPositionsDataPort.write(true);
+        pImpl->jointVelocitiesDataPort.write(true);
+        pImpl->comPositionDataPort.write(true);
+        pImpl->comVelocityDataPort.write(true);
+        pImpl->comProperAccelerationInBaseFrameDataPort.write(true);
+        pImpl->comProperAccelerationInWorldFrameDataPort.write(true);
+    }
+
+    if (pImpl->isWrenchProviderDevice) {
+
+        // Prepare wrench measurements source names data
+        yarp::os::Bottle& wrenchMeasurementSourceNamesYarpBottle = pImpl->wrenchMeasurementSourceNamesDataPort.prepare();
+        pImpl->dataBuffersConversionHelper.setBuffer(wrenchMeasurementSourceNamesYarpBottle, pImpl->wrenchMeasurementSourceNames);
+
+        // Prepare wrench measurement values data
+        yarp::sig::Vector& wrenchMeasurementValuesYarpVector = pImpl->wrenchMeasurementValuesDataPort.prepare();
+        pImpl->dataBuffersConversionHelper.setBuffer(wrenchMeasurementValuesYarpVector, pImpl->wrenchMeasurementValues);
+
+        // Send data through yarp ports
+        pImpl->wrenchMeasurementSourceNamesDataPort.write(true);
+        pImpl->wrenchMeasurementValuesDataPort.write(true);
+    }
+
+    if (pImpl->isDynamicsEstimatorDevice) {
+
+        // Prepare wrench estimates source names data
+        yarp::os::Bottle& wrenchEstimateSourceNamesYarpBottle = pImpl->wrenchEstimateSourceNamesDataPort.prepare();
+        pImpl->dataBuffersConversionHelper.setBuffer(wrenchEstimateSourceNamesYarpBottle, pImpl->wrenchEstimateSourceNames);
+
+        // Prepare wrench estimate values data
+        yarp::sig::Vector& wrenchEstimateValuesYarpVector = pImpl->wrenchEstimateValuesDataPort.prepare();
+        pImpl->dataBuffersConversionHelper.setBuffer(wrenchEstimateValuesYarpVector, pImpl->wrenchEstimateValues);
+
+        // Prepare dynamics joint names data
+        yarp::os::Bottle& dynamicsJointNamesYarpBottle = pImpl->dynamicsJointNamesDataPort.prepare();
+        pImpl->dataBuffersConversionHelper.setBuffer(dynamicsJointNamesYarpBottle, pImpl->dynamicsJointNames);
+
+        // Prepare joint torques data
+        yarp::sig::Vector& jointTorqesYarpVector = pImpl->jointTorquesDataPort.prepare();
+        pImpl->dataBuffersConversionHelper.setBuffer(jointTorqesYarpVector, pImpl->jointTorques);
+
+
+        // Send data through yarp ports
+        pImpl->wrenchEstimateSourceNamesDataPort.write(true);
+        pImpl->wrenchEstimateValuesDataPort.write(true);
+        pImpl->dynamicsJointNamesDataPort.write(true);
+        pImpl->jointTorquesDataPort.write(true);
+    }
+
+
+
+}
+
+bool HumanDataCollector::attach(yarp::dev::PolyDriver *poly)
+{
+    if (!poly) {
+        yError() << LogPrefix << "Passed PolyDriver device is nullptr";
+        return false;
+    }
+
+    // Get the passed device name
+    const std::string deviceName = poly->getValue("device").asString();
```

Comment:
I would check that "device" value exists before accessing it, I'm not sure what would happen otherwise.